lib/samlp.js

```diff
@@ -110,12 +110,16 @@ function getSamlResponse(samlConfig, user, callback) {
   }, function (err, samlAssertion) {
     if (err) return callback(err);
 
-    var SAMLResponse = buildSamlResponse(Object.assign({}, options, {
-      samlAssertion: samlAssertion,
-      samlStatusCode: options.samlStatusCode || constants.STATUS.SUCCESS
-    }));
-
-    callback(null, SAMLResponse);
+    try {
+      var SAMLResponse = buildSamlResponse(Object.assign({}, options, {
+        samlAssertion: samlAssertion,
+        samlStatusCode: options.samlStatusCode || constants.STATUS.SUCCESS
+      }));
+
+      callback(null, SAMLResponse);
+    } catch (err) {
+      callback(err);
+    }
```

Comment:
Could you modify this code to avoid the "double-callback error" scenario?

https://joseoncode.com/2013/12/27/case-of-double-callbacks/ TL;DR; move the `callback` inside the `try` after it, and add a `return` in front of the one in the `catch`

Reply:
Good point! 👍  Updated.